ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/WitnessesTable.scala

```diff
@@ -31,11 +31,14 @@ sealed abstract class WitnessesTable(tableName: String) {
       transactionId: TransactionId,
       witnesses: DisclosureRelation,
   ): Option[BatchSql] = {
-    if (witnesses.nonEmpty) {
-      val ws = DisclosureRelation
-        .flatten(witnesses)
-        .map { case (nodeId, party) => parameters(transactionId)(nodeId, party) }
-        .toSeq
+    val flattenedWitnesses = DisclosureRelation.flatten(witnesses)
+    if (flattenedWitnesses.nonEmpty) {
+      val ws = flattenedWitnesses.map {
+        case (nodeId, party) => parameters(transactionId)(nodeId, party)
+      }.toSeq
+      if (ws.isEmpty) {
+        sys.error(witnesses.toString)
```

Comment:
Not convinced `sys.error` is the most pleasant way to handle errors here. What does this even mean, and can we do something with it? Perhaps make it clear that things can error with an `Either` or a `Try`?

Reply:
Another leftover from debugging, sorry for the sloppy PR.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5029/commits/54ec136a17bf594b201a8fb6bd8c082e1d755f34